packages/passport-interface/src/api/requestEncryptedStorage.ts

```diff
@@ -0,0 +1,33 @@
+import { EncryptedPacket } from "@pcd/passport-crypto";
+import { LoadE2EERequest, LoadE2EEResponse } from "../RequestTypes";
+import { APIResult } from "./apiResult";
+import { POST } from "./constants";
+
+/**
+ * Never rejects. All information encoded in the resolved response.
+ */
+export async function requestEncryptedStorage(
+  passportServer: string,
+  blobKey: string
+): Promise<EncryptedStorageResult> {
+  const request: LoadE2EERequest = { blobKey };
+  const url = `${passportServer}/sync/load`;
+
+  try {
+    const res = await fetch(url, {
+      ...POST,
+      body: JSON.stringify(request)
+    });
+
+    if (!res.ok) {
+      return { error: "failed to load e2ee backup" };
```

Comment:
Trying to understand the pattern here: Are you losing information here about why this failed at deeper levels?  If I were debugging this would that info already be in a log somewhere?

Reply:
i'm still thinking about the best way to propagate erros

Reply:
Common problem.  I like Java's "cause exception" pattern personally, though it can get large and complex to read.  Not sure if there's a similar pattern in JavaScript.

Reply:
It is possible for JS exceptions to wrap other exceptions, so that a chain of them can be built up as the exception bubbles up the stack. In https://github.com/proofcarryingdata/zupass/pull/612 I refactored error-handling in the sync service to use this approach, and modified the `reportError` method in `RollbarService` to use the causal chain of exceptions when logging to Rollbar.

The thing that definitely doesn't work is catching an error at a low level, logging it, then returning a generic error result to the caller, which will log its own error but with no way for the logging service to tell that the two errors are connected. If the caller wants to log something, it also wants to see what the original exception was.